### PR TITLE
[Merged by Bors] - feat(AlgebraicGeometry): api for morphism properties local at the source and target

### PR DESCRIPTION
**Pull request by chrisflav**
A morphism property, that is both local at the source and the target and additionally is stable under post-composition with open immersions, can be checked locally around points of the source. This is what the [stacks](https://stacks.math.columbia.edu/tag/04QW) project calls "local on the source-and-target".

---

- [x] depends on: #17264

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/AlgebraicGeometry/Morphisms/Basic.lean
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/Morphisms/Basic.lean:282]:
```suggestion
lemma resLE [IsLocalAtTarget P] {U : Y.Opens} {V : X.Opens} (e : V ≤ f ⁻¹ᵁ U)
```
I think this is enough?

File: Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean
Comment (erdOne) [inline Mathlib/AlgebraicGeometry/Morphisms/RingHomProperties.lean:196]:
```suggestion
instance (priority := 900) : IsLocalAtTarget P := inferInstance
```

**Review by erdOne — APPROVED** [review]
LGTM modulo two minor comments

**Comment by jcommelin** [comment]
maintainer merge

**Pull request closed without merging**
